### PR TITLE
Add MAX32xxx counter RTC driver

### PR DESCRIPTION
**Pull request by oknshn**
This PR adds counter RTC driver for ADI MAX32xxx series microcontrollers. Counter tests are also enabled.

### CODE REVIEW

**Comment by oknshn** [comment]
> Needs rebase
Done.

File: boards/adi/max32655evkit/max32655evkit_max32655_m4.dts [hunk 1]
```diff
@@ -78,6 +78,13 @@
 	status = "okay";
 };
 
+/*
+ * ERTCO requires for counter RTC
```
Comment (ttmut) [inline boards/adi/max32655evkit/max32655evkit_max32655_m4.dts:82]:
```suggestion
 * ERTCO is required for counter RTC
```

**Review by ttmut — CHANGES_REQUESTED** [review]
Not a hard requirement but you may consider merging commits to `boards` from same author into a single one to keep commit history brief.

**Comment by oknshn** [comment]
> Not a hard requirement but you may consider merging commits to `boards` from same author into a single one to keep commit history brief.

Done

**Review by ttmut — APPROVED** [review]

**Review by ozersa — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]
looks good, I like the `rtc_counter` name, its always confusing with RTC, is it real time clock, is it real time counter, or RTCC, is that real time clock calendar or real time clock counter :D

**Comment by oknshn** [comment]
Hello @nordic-krch,
Could you take a look?

**Comment by oknshn** [comment]
> looks good, I like the `rtc_counter` name, its always confusing with RTC, is it real time clock, is it real time counter, or RTCC, is that real time clock calendar or real time clock counter :D
Thanks, It is `real time counter`, I can create follow-up PR for it.

**Review by MaureenHelm — APPROVED** [review]

**Comment by oknshn** [comment]
> Hello @nordic-krch, Could you take a look?
Hello @nordic-krch,
Waiting for your review.

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**